frontend/src/hooks/api/aliases.ts

```diff
@@ -0,0 +1,150 @@
+export type CustomAliasCreateFn = (address: string) => Promise<void>;
+export type AliasUpdateFn = (alias: Partial<AliasData> & { id: number }) => Promise<void>;
+export type AliasDeleteFn = (id: number) => Promise<void>;
+type WithRandomAliasCreater = {
+  create: RandomAliasCreateFn;
+};
+type WithCustomAliasCreater = {
+  create: CustomAliasCreateFn;
+};
+type WithUpdater = {
+  update: AliasUpdateFn;
+};
+type WithDeleter = {
+  delete: AliasDeleteFn;
+};
+
+export function useAliases(): {
+  randomAliasData: SWRResponse<RandomAliasData[], unknown> & WithRandomAliasCreater & WithUpdater & WithDeleter;
+  customAliasData: SWRResponse<CustomAliasData[], unknown> & WithCustomAliasCreater & WithUpdater & WithDeleter;
+} {
+  const randomAliases: SWRResponse<RandomAliasData[], unknown> =
+    useApiV1("/relayaddresses/");
+  const customAliases: SWRResponse<CustomAliasData[], unknown> =
+    useApiV1("/domainaddresses/");
+
+  const randomAliasCreater: RandomAliasCreateFn = async () => {
+    await apiFetch("/relayaddresses/", {
+      method: "POST",
+      body: JSON.stringify({ enabled: true }),
+    });
+    randomAliases.mutate();
+  }
+  const customAliasCreater: CustomAliasCreateFn = async (address) => {
+    await apiFetch("/domainaddresses/", {
+      method: "POST",
+      body: JSON.stringify({ enabled: true, address: address }),
+    });
+    customAliases.mutate();
+  }
+
+  const getUpdater: (type: "random" | "custom") => AliasUpdateFn = (type) => {
+    return async (aliasData) => {
+      const endpoint =
+        type === "random"
+          ? `/relayaddresses/${aliasData.id}/`
+          : `/domainaddresses/${aliasData.id}/`;
+
+      await apiFetch(endpoint, {
+        method: "PATCH",
+        body: JSON.stringify(aliasData),
+      });
+      if (type === "random") {
+        randomAliases.mutate();
+      } else {
+        customAliases.mutate();
+      }
+    };
+  };
+
+  const getDeleter: (type: "random" | "custom") => AliasDeleteFn = (type) => {
+    return async (aliasId) => {
+      const endpoint =
+        type === "random"
+          ? `/relayaddresses/${aliasId}/`
+          : `/domainaddresses/${aliasId}/`;
+
+      await apiFetch(endpoint, {
+        method: "DELETE",
+      });
+      if (type === "random") {
+        randomAliases.mutate();
+      } else {
+        customAliases.mutate();
+      }
+    };
+  };
+
+  const randomAliasData = {
+    ...randomAliases,
+    create: randomAliasCreater,
+    update: getUpdater("random"),
+    delete: getDeleter("random"),
+  };
+  const customAliasData = {
+    ...customAliases,
+    create: customAliasCreater,
+    update: getUpdater("custom"),
+    delete: getDeleter("custom"),
+  };
+
+  return {
+    randomAliasData: randomAliasData,
+    customAliasData: customAliasData,
+  };
+}
+
+export function isRandomAlias(alias: AliasData): alias is RandomAliasData {
+  return typeof (alias as RandomAliasData).generated_for === "string";
```

Comment:
question: Is this saying that an alias must have a `string` `generated_for` value to be a `RandomAlias`? That won't always be the case - there are random aliases with null or empty `generated_for`.

question: is this the right way to check if an `alias` is a random alias or not? Is there a way to just check the type of `alias` is `RandomAliasData` ?

Reply:
Yes, so that's a great callout! Unfortunately at runtime we only have JS types available, which for these is just `object`, so I added d343c78d8a680dc2ed8de8ac9c22cf6dd544e1bd to add an explicit type marker. We'll probably want to add something similar if we add an API that returns both types of addresses in a single request.